L1Trigger/L1TGlobal/src/L1TGlobalUtil.cc

```diff
@@ -0,0 +1,529 @@
+////
+/// \class l1t::L1TGlobalUtil.cc
+///
+/// Description: Dump Accessors for L1 GT Result.
+///
+/// Implementation:
+///    
+///
+/// \author: Brian Winer Ohio State
+///
+/// 
+#include "L1Trigger/L1TGlobal/interface/L1TGlobalUtil.h"
+
+#include <iostream>
+#include <fstream>
+
+#include "CondFormats/DataRecord/interface/L1TGlobalTriggerMenuRcd.h"
+
+#include "FWCore/Framework/interface/Event.h"
+#include "FWCore/ParameterSet/interface/ParameterSet.h"
+#include "FWCore/Utilities/interface/EDGetToken.h"
+#include "FWCore/Utilities/interface/InputTag.h"
+#include "FWCore/Framework/interface/EventSetup.h"
+#include "FWCore/Framework/interface/Frameworkfwd.h"
+#include "FWCore/Framework/interface/ESHandle.h"
+
+
+#include "FWCore/MessageLogger/interface/MessageLogger.h"
+#include "FWCore/MessageLogger/interface/MessageDrop.h"
+
+
+// constructor
+l1t::L1TGlobalUtil::L1TGlobalUtil(std::string preScaleFileName, unsigned int psColumn) 
+{
+
+    // initialize cached IDs
+    m_l1GtMenuCacheID = 0ULL;
+
+    m_filledPrescales = false;
+
+    m_preScaleFileName = preScaleFileName;
+
+    m_numberPhysTriggers = 512; //need to get this out of the EventSetup
+
+    m_PreScaleColumn = psColumn;
+
+}
+
+// destructor
+l1t::L1TGlobalUtil::~L1TGlobalUtil() {
+ 
+}
+
+
+void l1t::L1TGlobalUtil::retrieveL1(const edm::Event& iEvent, const edm::EventSetup& evSetup,
+                                    edm::EDGetToken gtAlgToken) {
+
+// get / update the trigger menu from the EventSetup
+// local cache & check on cacheIdentifier
+    unsigned long long l1GtMenuCacheID = evSetup.get<L1TGlobalTriggerMenuRcd>().cacheIdentifier();
+
+    if (m_l1GtMenuCacheID != l1GtMenuCacheID) {
+
+        //std::cout << "Attempting to get the Menu " << std::endl;
+        edm::ESHandle< TriggerMenu> l1GtMenu;
+        evSetup.get< L1TGlobalTriggerMenuRcd>().get(l1GtMenu) ;
+        m_l1GtMenu =  l1GtMenu.product();
+       //(const_cast<TriggerMenu*>(m_l1GtMenu))->buildGtConditionMap();
```

Comment:
@mulhearn m_l1GtMenu doesn't need to be a member data. (it is only consumed by m_algorithmMap).

But that asked, what is the cpu benefit of caching this?